vnext/Microsoft.ReactNative/Views/ReactRootControl.cpp

```diff
@@ -607,6 +608,53 @@ void ReactRootControl::ReloadViewHost() noexcept {
   }
 }
 
+void ReactRootControl::AttachBackHandlers(XamlView const &rootView) noexcept {
+  auto rootElement(rootView.as<winrt::UIElement>());
+  if (rootElement == nullptr) {
+    assert(false);
+    return;
+  }
+
+  winrt::Windows::UI::Core::SystemNavigationManager::GetForCurrentView().BackRequested(
+      [this](winrt::IInspectable const & /*sender*/, winrt::BackRequestedEventArgs const &args) {
+        args.Handled(OnBackRequested());
+      });
+
+  // Handle keyboard "back" button press
+  winrt::KeyboardAccelerator goBack{};
+  goBack.Key(winrt::VirtualKey::GoBack);
+  goBack.Invoked(
+      [this](winrt::KeyboardAccelerator const & /*sender*/, winrt::KeyboardAcceleratorInvokedEventArgs const &args) {
+        args.Handled(OnBackRequested());
+      });
+  rootElement.KeyboardAccelerators().Append(goBack);
+
+  // Handle Alt+Left keyboard shortcut
+  winrt::KeyboardAccelerator altLeft{};
+  altLeft.Key(winrt::VirtualKey::Left);
+  altLeft.Invoked(
+      [this](winrt::KeyboardAccelerator const & /*sender*/, winrt::KeyboardAcceleratorInvokedEventArgs const &args) {
+        args.Handled(OnBackRequested());
+      });
+  rootElement.KeyboardAccelerators().Append(altLeft);
+  altLeft.Modifiers(winrt::VirtualKeyModifiers::Menu);
+}
+
+bool ReactRootControl::OnBackRequested() noexcept {
+  if (m_weakReactInstance.IsExpired()) {
+    return false;
+  }
+
+  auto &legacyReactInstance = query_cast<Mso::React::ILegacyReactInstance &>(*m_weakReactInstance.GetStrongPtr());
+  auto uwpReactInstance = legacyReactInstance.UwpReactInstance();
```

Comment:
`ILegacyReactInstance` should have its own `CallJsFunction` function on it. (This code is definitely a bit of a mess).